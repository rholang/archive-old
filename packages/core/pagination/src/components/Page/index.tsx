import React, { Component } from 'react';
import Button, { ButtonProps, ButtonAppearances } from '@atlaskit/button';

type Diff<T, U> = T extends U ? never : T;

type PagePropsType = Diff<
  ButtonProps,
  {
    appearance?: ButtonAppearances;
    autoFocus: boolean;
    isDisabled: boolean;
    isLoading: boolean;
    spacing: 'compact' | 'default' | 'none';
    shouldFitContainer: boolean;
    type: 'button' | 'submit';
  }
> & { page?: any };

export default class Page extends Component<PagePropsType> {
  render() {
    return <Button {...this.props} appearance="subtle" />;
  }
}
