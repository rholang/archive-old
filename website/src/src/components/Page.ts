import styled from 'styled-components';
import { gridSize, math } from '@atlaskit/theme';

const PageContainer = styled.main`
  width: 100vw;
  max-width: calc(640px + 4rem);
  box-sizing: border-box;
  margin: 2rem auto;
  padding: 0 2rem;
`;

export default PageContainer;

export const Title = styled.h1`
  margin-bottom: 1em;
`;

export const Section = styled.section`
  margin-top: 3em;

  p {
    line-height: 1.4em;
  }
`;

export const Intro = styled.p`
  font-size: ${math.multiply(gridSize, 2)}px;
  font-weight: 300;
  line-height: 1.4em;
`;
