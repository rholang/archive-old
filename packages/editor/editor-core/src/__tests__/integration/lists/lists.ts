import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  getDocFromElement,
  fullpage,
  editable,
  getProseMirrorPos,
} from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { KEY } from '../../__helpers/page-objects/_keyboard';

import floatsAdf from './__fixtures__/lists-adjacent-floats.adf.json';

const PM_FOCUS_SELECTOR = '.ProseMirror-focused';

async function insertList(
  page: any,
  modifierKey: string,
  list: 'number' | 'bullet',
) {
  const listKey = list === 'number' ? '7' : '8';
  await page.browser.keys([modifierKey, KEY.SHIFT, listKey]);
  await page.browser.keys([modifierKey, KEY.SHIFT]); // release modifier keys
  await page.type(editable, 'item');
  await page.browser.keys(['Enter', 'Enter']); // double enter to exit list
}

BrowserTestCase(
  `list: shouldn't change focus on tab if the list is not indentable`,
  { skip: ['ie'] },
  async (client: any, testName: string) => {
    const page = new Page(client);
    await page.goto(fullpage.path);
    await page.waitForSelector(fullpage.placeholder);
    await page.click(fullpage.placeholder);

    // Investigate why string based input (without an array) fails in firefox
    // https://product-fabric.atlassian.net/browse/ED-7044
    await page.type(editable, '* '.split(''));
    await page.type(editable, 'abc');
    await page.keys('Return');
    await page.keys('Tab');
    await page.type(editable, '123');
    await page.keys('Tab');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
    expect(await page.isExisting(PM_FOCUS_SELECTOR)).toBeTruthy();
  },
);

BrowserTestCase(
  'list: should be able to insert lists via keyboard shortcut (Windows)',
  { skip: ['ie', 'edge', 'safari', 'firefox'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, { appearance: 'full-page', allowLists: true });
    await page.click(editable);
    await insertList(page, KEY.CONTROL, 'number');
    await insertList(page, KEY.CONTROL, 'bullet');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'list: should be able to insert lists via keyboard shortcut (Mac)',
  { skip: ['ie', 'edge', 'chrome', 'firefox'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, { appearance: 'full-page', allowLists: true });
    await page.click(editable);
    await insertList(page, KEY.META, 'number');
    await insertList(page, KEY.META, 'bullet');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

/**
 * An example of what our page looks like
 * We use the arrow keys to navigate this list
 * In Firefox we want to ensure our cursor doesnt
 * jump outside the list on the first navigation
 * movement, but goes up and down the list as expected.
 *
 * +-----------------+   This is text
 * |                 |   1. One
 * |                 |     2. Two
 * |   float:left;   |       3. Three
 * |                 |     4. Four
 * |                 |   5. Five
 * +-----------------+
 */
BrowserTestCase(
  'list: should be able to navigate lists correctly in firefox',
  { skip: ['ie', 'edge', 'chrome', 'safari'] },
  async (client: any) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: 'full-page',
      media: {
        allowMediaSingle: true,
      },
      allowLists: true,
      defaultValue: floatsAdf,
      shouldFocus: true,
    });

    // These loops navigate up and down the lists
    // We want to ensure we remain in the list.
    for (let i = 0; i < 11; i++) {
      await page.keys('ArrowDown');
    }

    for (let i = 0; i < 5; i++) {
      await page.keys('ArrowRight');
    }

    for (let i = 0; i < 3; i++) {
      await page.keys('ArrowUp');
    }

    const pos = await getProseMirrorPos(page);
    // Start of the word Six
    expect(pos).toEqual(217);
  },
);
