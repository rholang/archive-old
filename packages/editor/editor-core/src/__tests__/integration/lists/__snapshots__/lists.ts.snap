// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list: should be able to insert lists via keyboard shortcut (Mac) 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "item",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "item",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`list: should be able to insert lists via keyboard shortcut (Windows) 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "item",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "item",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`list: shouldn't change focus on tab if the list is not indentable 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "abc",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "123",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "listItem",
                },
              ],
              "type": "bulletList",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;
